src/PerpetualStableSwap.sol

```diff
@@ -0,0 +1,137 @@
+    IERC20 public immutable tokenB;
+    uint256 public immutable halfSpreadBps;
+    address public immutable target;
+    bytes32 public domainSeparator;
+
+    // There are 10k basis points in a unit
+    uint256 public constant BPS = 10_000;
+
+    constructor(
+        IERC20 _tokenA,
+        IERC20 _tokenB,
+        uint256 _halfSpreadBps,
+        address _target,
+        GPv2Settlement _settlementContract
+    ) {
+        tokenA = _tokenA;
+        tokenB = _tokenB;
+        halfSpreadBps = _halfSpreadBps;
+        domainSeparator = _settlementContract.domainSeparator();
+
+        if (_target == address(0)) {
+            _target = address(this);
+            // Otherwise we expect the target to set allowance itself
+            _tokenA.approve(
+                address(_settlementContract.vaultRelayer()),
+                uint(-1)
+            );
+            _tokenB.approve(
+                address(_settlementContract.vaultRelayer()),
+                uint(-1)
+            );
+        }
+        target = _target;
+        emit ConditionalOrderCreated(_target);
+    }
+
+    function getTradeableOrder()
+        external
+        view
+        override
+        returns (GPv2Order.Data memory)
+    {
+        uint256 balanceA = tokenA.balanceOf(target);
+        uint256 balanceB = tokenB.balanceOf(target);
+
+        // Always sell whatever of the two tokens we have more of
+        IERC20 sellToken;
+        IERC20 buyToken;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        if (convertAmount(tokenA, balanceA, tokenB) > balanceB) {
+            sellToken = tokenA;
+            buyToken = tokenB;
+            sellAmount = balanceA;
+            buyAmount = convertAmount(tokenA, balanceA, tokenB)
+                .mul(BPS.add(halfSpreadBps))
+                .div(BPS);
+        } else {
+            sellToken = tokenB;
+            buyToken = tokenA;
+            sellAmount = balanceB;
+            buyAmount = convertAmount(tokenB, balanceB, tokenA)
+                .mul(BPS.add(halfSpreadBps))
+                .div(BPS);
+        }
+        require(sellAmount > 0, "not funded");
+
+        // Unless spread is 0 (and there is no surplus), order collision is not an issue as sell and buy amounts should
+        // increase for each subsequent order. We therefore set validity to a large time span
+        // Note, that reducing currenbt block to a common start time is needed so that the order returned here
+        // does not change between the time it is queried and the time it is settled.
+        uint32 validity = 1 weeks;
+        // solhint-disable-next-line not-rely-on-time
+        uint32 currentTimeBucket = ((uint32(block.timestamp) / validity) + 1) *
+            validity;
+        return
+            GPv2Order.Data(
+                sellToken,
+                buyToken,
+                target,
+                sellAmount,
+                buyAmount,
+                currentTimeBucket + validity,
```

Comment:
This can be up to two weeks. From the comments, it doesn't seem intentional (although it makes sense, otherwise you can return orders that are only valid for a few seconds).

Reply:
Clarified the comment